p2p/upgrade.go

```diff
@@ -78,7 +78,7 @@ func isBootnode(h host.Host, bootnodes []string) (bool, error) {
 }
 
 // Upgrade creates Host instance from host.Host.
-func Upgrade(h host.Host, opts ...Opt) (*Host, error) {
+func Upgrade(h host.Host, dataDir string, opts ...Opt) (*Host, error) {
```

Comment:
Well, I don't think it could. The `type Opt func(fh *Host)` gets a pointer to the `Host` instance and can do something with it (i.e. set a field). However, `dataDir` is not, and needs not to be, a part of `Host` (there is no field `dataDir` in `Host` because there is no need for it).

Reply:
before the change `dataDir` was a field of the host and its purpose is to be passed along all the way down to the addressbook. It is also used in a call to `EnsureIdentity`.

I think it could be named `identitiesDir` instead and then be part of the hosts config, just as it was before just with a clearer name of its intended purpose.

Reply:
Yes, it was a field, but it doesn't need to be as it is used one time during the initialization.

Reply:
I think as a caller of the Upgrade function it is confusing to a) not know what `dataDir` is used for (which data is stored in that dir?) and b) having to set it explicitly when other configuration is set via opts and/or the `Host` object.